enhancements/20231113-witness-node-support.md

```diff
@@ -0,0 +1,166 @@
+# Support witness node
+
+## Summary
+
+The witness node is a lightweight node that only runs the control plane components. The witness node is not schedulable and does not run any workload. The witness node forms a quorum with the other two nodes.
+
+### Related Issues
+
+https://github.com/harvester/harvester/issues/3266
+
+## Motivation
+
+Some users do not have three powerful nodes for the edge uses case. Instead, they would have two powerful nodes and one lightweight node. The lightweight node is only used for the `etcd` role. Kubernetes need at least three nodes to form a quorum. The witness node support will help the edge uses case. That will benefit both cost and high availability.
+
+### Goals
+
+Create a three-node cluster with one witness node.
+
+## Proposal
+
+### User Stories
+
+For the edge use case, the witness node support benefits both cost and high availability. The classical scenario would be two nodes and one witness node.
+
+### User Experience In Detail
+
+#### Installation
+- Install with interactive mode:
+   1. There is no change for the first node (`create` mode). The first node must be the master node.
+   2. For the second/third node, we will select the `join` mode to join the current cluster. Then, we will move to the role selection panel. On this panel, we can choose the role of `Default` or `Etcd`. If we select the `Etcd` role, the node will be promoted to only the etcd role, which will be the witness node.
```

Comment:
@Vicente-Cheng should we update the role to `Witness`?

Reply:
rest looks good and i can approve this once this is updated.

Reply:
Updated!